### PR TITLE
Miawong/add typechecking version history

### PR DESCRIPTION
**Pull request by miaawong**
#### Which issue(s) this PR fixes:

Fixes # [sc-58965]

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/components/shared/modals/HelmDeployModal.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ function makeDeployCommand({
 }: {
   appSlug: string;
   chartPath: string;
-  revision: string | null;
+  revision: number | null;
   showDownloadValues: boolean;
```
Comment (miaawong) [inline web/src/components/shared/modals/HelmDeployModal.tsx:19]: super confused as to if this is string or number but it errors out if it's a string.

File: web/src/components/apps/AppVersionHistory.tsx [hunk 2]
```diff
@@ -1092,46 +1220,52 @@ class AppVersionHistory extends Component {
...
-          (release) => release.parentSequence !== selectedRelease.parentSequence
+          (release: Version) =>
+            release.parentSequence !== selectedRelease.parentSequence
         ),
       });
     }
   };
 
-  displayTooltip = (key, value) => {
-    return () => {
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1120]: not being called anywhere

File: web/src/components/apps/AppVersionHistory.tsx [hunk 3]
```diff
@@ -1477,20 +1618,28 @@ class AppVersionHistory extends Component {
               {({
                 download,
                 clearError: clearDownloadError,
-                error: downloadError,
-                isDownloading,
+                downloadError: downloadError,
+                // isDownloading,
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1622]: isDownloading isn't being used here

File: web/src/components/apps/AppVersionHistory.tsx [hunk 4]
```diff
@@ -1839,7 +2015,6 @@ class AppVersionHistory extends Component {
                             isBundleUploading={this.props.isBundleUploading}
                             checkingUpdateText={checkingUpdateMessage}
                             checkingUpdateTextShort={checkingUpdateTextShort}
-                            noUpdatesAvalable={this.props.noUpdatesAvalable}
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1842]: not being used here

File: web/src/components/apps/AppVersionHistory.tsx [hunk 5]
```diff
@@ -278,30 +406,31 @@ class AppVersionHistory extends Component {
...
+  onUploadProgress = (progress: number, size: number, resuming = false) => {
     this.setState({
       uploadProgress: progress,
       uploadSize: size,
       uploadResuming: resuming,
     });
   };
 
-  onUploadError = (message) => {
+  onUploadError = (message: String) => {
```
Comment (rianfowler) [inline web/src/components/apps/AppVersionHistory.tsx:425]:
```suggestion
  onUploadError = (message: string) => {
```
does this work?

File: web/src/components/apps/AppVersionHistory.tsx [hunk 6]
```diff
@@ -676,10 +804,10 @@ class AppVersionHistory extends Component {
     });
   };
 
-  renderVersionDownloadStatus = (version) => {
+  renderVersionDownloadStatus = (version: Version) => {
     const { versionDownloadStatuses } = this.state;
-
-    if (!versionDownloadStatuses.hasOwnProperty(version.sequence)) {
+    //!versionDownloadStatuses.hasOwnProperty(version.sequence)
```
Comment (rianfowler) [inline web/src/components/apps/AppVersionHistory.tsx:809]:
```suggestion
```

File: web/src/components/shared/modals/HelmDeployModal.tsx (hunk 1 above)
Comment (rianfowler) [reply web/src/components/shared/modals/HelmDeployModal.tsx:19]: we'll clean it up later. i think we'll clean up the response types better once we've done the initial typing

**Review by rianfowler — APPROVED** [review]

**Review by rianfowler — DISMISSED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request closed without merging**